plugins/eddn.py

```diff
@@ -633,6 +631,277 @@ def export_journal_entry(self, cmdr: str, is_beta: bool, entry: Mapping[str, Any
+        :param entry: the journal entry to send
+        """
+        # {
+        #   "timestamp":"2021-09-26T11:55:03Z",
+        #   "event":"ScanBaryCentre",
+        #   "StarSystem":"Khruvani",
+        #   "SystemAddress":13864557159793,
+        #   "BodyID":21,
+        #   "SemiMajorAxis":863683605194.091797,
+        #   "Eccentricity":0.001446,
+        #   "OrbitalInclination":-0.230714,
+        #   "Periapsis":214.828581,
+        #   "OrbitalPeriod":658474677.801132,
+        #   "AscendingNode":21.188568,
+        #   "MeanAnomaly":208.765388
+        # }
+        #######################################################################
+        # Elisions
+        entry = filter_localised(entry)
+        #######################################################################
+
+        #######################################################################
+        # Augmentations
+        #######################################################################
+        ret = this.eddn.entry_augment_system_data(entry, entry['StarSystem'])
+        if isinstance(ret, str):
+            return ret
+
+        entry = ret
+        #######################################################################
+
+        msg = {
+            '$schemaRef': f'https://eddn.edcd.io/schemas/scanbarycentre/1{"/test" if is_beta else ""}',
+            'message': entry
+        }
+
+        this.eddn.export_journal_entry(cmdr, entry, msg)
+        return None
+
+    def export_journal_navroute(
+            self, cmdr: str, is_beta: bool, entry: Dict[str, Any]
+    ) -> Optional[str]:
+        """
+        Send a NavRoute to EDDN on the correct schema.
+
+        :param cmdr: the commander under which this upload is made
+        :param is_beta: whether or not we are in beta mode
+        :param entry: the journal entry to send
+        """
+        # {
+        #    "timestamp":"2021-09-24T14:33:15Z",
+        #    "event":"NavRoute",
+        #    "Route":[
+        #       {
+        #         "StarSystem":"Ross 332",
+        #         "SystemAddress":3657198211778,
+        #         "StarPos":[-43.62500,-23.15625,-74.12500],
+        #         "StarClass":"K"
+        #       },
+        #       {
+        #          "StarSystem":"BD+44 1040",
+        #          "SystemAddress":2832564490946,
+        #          "StarPos":[-31.56250,1.84375,-92.37500],
+        #          "StarClass":"K"
+        #       },
+        #       {
+        #         "StarSystem":"Aonga",
+        #         "SystemAddress":6680855188162,
+        #         "StarPos":[-34.46875,9.53125,-90.87500],
+        #         "StarClass":"M"
+        #       }
+        #    ]
+        #  }
+        #######################################################################
+        # Elisions
+        #######################################################################
+        # This will reject with the Odyssey flag present
```

Comment:
NB: We're discussing allowing this in the schema, probably along with 'horizons'.

Reply:
can we add a `WORKAROUND` or maybe a (new) `NOTE` comment here for later easy grepping? or I guess `XXX` might work?